### PR TITLE
Added example using JAX and the scipy interface.

### PR DESCRIPTION
**Pull request by jhelgert**
As we all know, calculating derivatives by hand can be quite error-prone. During the last months, I often used the [JAX](https://jax.readthedocs.io/en/latest/index.html) library to evaluate gradients and (dense) jacobians via AD (algorithmic differentiation), so I thought it might be an interesting example.

### CODE REVIEW

**Comment by moorepants** [comment]
This is nice use of JAX. I think the example would get more visibility if added to the documentation. Would you be willing to move it there? You'll probably have to add jax as a dependency in the CI for the docs to build.

**Comment by jhelgert** [comment]
@moorepants Done. Is it really necessary to add jax as a dependency for docs? I am quite new to sphinx and I don't really see why it might be needed.

**Comment by moorepants** [comment]
> Is it really necessary to add jax as a dependency for docs?

It may not be, if the examples aren't run or doc-tested.

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -26,15 +26,26 @@ the same behaviour as ``scipy.optimize.minimize``, for example::
    success: True
          x: array([1., 1., 1., 1., 1.])
 
-Problem Interface
-=================
+Algorithmic Differentation
+--------------------
```
Comment (moorepants) [inline docs/source/tutorial.rst:30]:
```
/home/moorepants/src/cyipopt/docs/source/tutorial.rst:30: WARNING: Title underline too short.

Algorithmic Differentation
--------------------
```

**Comment by moorepants** [comment]
Thanks!

I guess our CI is no longer running...
